pyanaconda/modules/storage/partitioning/interactive_utils.py

```diff
@@ -284,19 +286,151 @@ def validate_label(label, fmt):
+
+    if path in system_mount_points:
+        return _("That mount point is invalid. Try something else?")
+
+    if ((len(path) > 1 and path.endswith("/")) or
+            not path.startswith("/") or
+            " " in path or
+            re.search(r'/\.*/', path) or
+            re.search(r'/\.+$', path)):
+        # - does not end with '/' unless mountpoint _is_ '/'
+        # - starts with '/' except for "swap", &c
+        # - does not contain spaces
+        # - does not contain pairs of '/' enclosing zero or more '.'
+        # - does not end with '/' followed by one or more '.'
+        return _("That mount point is invalid. Try something else?")
+
+    return None
+
+
+def validate_raid_level(raid_level, num_members):
+    """Validate the given raid level.
+
+    :param raid_level: a RAID level
+    :param num_members: a number of members
+    :return: an error message
+    """
+    if num_members < raid_level.min_members:
+        return _("The RAID level you have selected (%(level)s) requires more "
+                 "disks (%(min)d) than you currently have selected (%(count)d).") \
+               % {
+                   "level": raid_level,
+                   "min": raid_level.min_members,
+                   "count": num_members
+               }
+
+    return None
+
+
+def validate_device_info(storage, dev_info, reformat):
+    """Validate the given device info.
+
+    :param storage: an instance of Blivet
+    :param dev_info: a device info to validate
+    :param reformat: is reformatting enabled?
+    :return: an error message
+    """
+    device = dev_info["device"]
+    disks = dev_info["disks"]
+    device_type = dev_info["device_type"]
+    fs_type = dev_info["fstype"]
+    encrypted = dev_info["encrypted"]
+    raid_level = dev_info["raid_level"]
+    mount_point = dev_info["mountpoint"]
+
+    changed_label = dev_info["label"] != getattr(device.format, "label", "")
+    changed_fstype = dev_info["fstype"] != device.format.type
+
+    if changed_label or changed_fstype:
+        error = validate_label(
+            dev_info["label"],
+            get_format(fs_type)
+        )
+        if error:
+            return error
+
+    changed_mount_point = dev_info["mountpoint"] != getattr(device.format, "mountpoint")
+
+    if reformat and not mount_point:
+        return _("Please enter a mount point.")
+
+    if changed_mount_point and mount_point:
+        error = validate_mount_point(
+            mount_point,
+            storage.mountpoints.keys()
+        )
+        if error:
+            return error
+
+    supported_types = (devicefactory.DEVICE_TYPE_PARTITION, devicefactory.DEVICE_TYPE_MD)
+
+    if mount_point == "/boot/efi" and device_type not in supported_types:
+        return _("/boot/efi must be on a device of type %(type)s or %(another)s") % {
+            "type": _(DEVICE_TEXT_MAP[devicefactory.DEVICE_TYPE_PARTITION]),
+            "another": _(DEVICE_TEXT_MAP[devicefactory.DEVICE_TYPE_MD])
+        }
```

Comment:
Please change this into the string `format` solution. The same works for the strings below.

Reply:
Fixed.